components/tidb_query/src/codec/mysql/time/datetime.rs

```diff
@@ -0,0 +1,1243 @@
+            strict_mode: sql_mode.contains(SqlMode::STRICT_ALL_TABLES),
+            no_zero_in_date: sql_mode.contains(SqlMode::NO_ZERO_IN_DATE),
+            no_zero_date: sql_mode.contains(SqlMode::NO_ZERO_DATE),
+            allow_invalid_date: sql_mode.contains(SqlMode::INVALID_DATES),
+            ignore_truncate: flags.contains(Flag::IGNORE_TRUNCATE),
+            time_zone: Some(ctx.cfg.tz.clone()),
+        }
+    }
+}
+
+impl From<DateTimeConfig> for EvalContext {
+    fn from(config: DateTimeConfig) -> EvalContext {
+        let mut eval_config = EvalConfig::new();
+        let mut sql_mode = SqlMode::empty();
+        let mut flags = Flag::empty();
+        eval_config.tz = config.time_zone.unwrap_or_else(Tz::utc);
+
+        if config.strict_mode {
+            sql_mode |= SqlMode::STRICT_ALL_TABLES;
+        }
+        if config.allow_invalid_date {
+            sql_mode |= SqlMode::INVALID_DATES;
+        }
+        if config.no_zero_date {
+            sql_mode |= SqlMode::NO_ZERO_DATE;
+        }
+        if config.no_zero_in_date {
+            sql_mode |= SqlMode::NO_ZERO_IN_DATE;
+        }
+
+        if config.ignore_truncate {
+            flags |= Flag::IGNORE_TRUNCATE;
+        }
+
+        eval_config.set_sql_mode(sql_mode).set_flag(flags);
+
+        EvalContext::new(Arc::new(eval_config))
+    }
+}
+
+#[derive(PartialEq, Clone, Copy, Debug)]
+pub enum TimeType {
+    Date,
+    DateTime,
+    TimeStamp,
+}
+
+impl TryFrom<FieldTypeTp> for TimeType {
+    type Error = codec::Error;
+    fn try_from(time_type: FieldTypeTp) -> Result<TimeType> {
+        Ok(match time_type {
+            FieldTypeTp::Date => TimeType::Date,
+            FieldTypeTp::DateTime => TimeType::DateTime,
+            FieldTypeTp::Timestamp => TimeType::TimeStamp,
+            _ => return Err(box_err!("Time does not support field type {}", time_type)),
+        })
+    }
+}
+
+// The common set of methods for `date/time`
+impl Time {
+    /// Returns the hour number from 0 to 23.
+    #[inline]
+    pub fn hour(self) -> u32 {
+        self.get_hour()
+    }
+
+    /// Returns the minute number from 0 to 59.
+    #[inline]
+    pub fn minute(self) -> u32 {
+        self.get_minute()
+    }
+
+    /// Returns the second number from 0 to 59.
+    #[inline]
+    pub fn second(self) -> u32 {
+        self.get_second()
+    }
+
+    /// Returns the number of microseconds since the whole second.
+    pub fn micro(self) -> u32 {
+        self.get_micro()
+    }
+
+    /// Returns the year number
+    pub fn year(self) -> u32 {
+        self.get_year()
+    }
+
+    /// Returns the month number
+    pub fn month(self) -> u32 {
+        self.get_month()
+    }
+
+    /// Returns the day number
+    pub fn day(self) -> u32 {
+        self.get_day()
+    }
+}
+
+// Parser
+impl Time {
+    // Split `input` with `.`, returns the whole part and a optional fractional part.
+    fn split_frac(input: &[u8]) -> Result<(&[u8], Option<&[u8]>)> {
```

Comment:
I think should use `&str` as param type but not `&[u8]`, becase the `pub` function `parse`'s param is `&str` but not `&[u8]`